cdci_data_analysis/analysis/parameters.py

```diff
@@ -397,77 +401,38 @@ def matches_owl_uri(cls, owl_uri: str) -> bool:
     @classmethod
     def from_owl_uri(cls, 
                      owl_uri, 
-                     value=None,
-                     units=None,
-                     name = None,
-                     allowed_units=None,
-                     default_units=None,
-                     units_name=None,
- 
-                     par_format=None,
-                     par_default_format=None,
-                     par_format_name=None,
- 
-                     default_type=None,
-                     allowed_types=None,
- 
-                     check_value=None,
-                     allowed_values=None):
+                     **kwargs):
         # TODO: what about units?
 
         parameter = None
-        
+
         for x in subclasses_recursive(cls):
             logger.debug("searching for class with owl_uri=%s, found %s", owl_uri, x)
             if x.matches_owl_uri(owl_uri):
                 logger.info("will construct %s by url %s", x, owl_uri)
+                call_kwargs = kwargs.copy()
+                call_signature = signature(x)
+                for par_name in kwargs.keys():
+                    if par_name not in call_signature.parameters:
+                        logger.warning("parameter %s, with value %s, is not used to construct a %s object, therefore this will be discarded for the instantiation", par_name, kwargs[par_name], x)
+                        call_kwargs.pop(par_name, None)                                                            
                 try:
-                    parameter = x(value=value,
-                            units=units,
-                            name=name,
-                            allowed_units=allowed_units,
-                            default_units=default_units,
-                            units_name=units_name,
-        
-                            par_format=par_format,
-                            par_default_format=par_default_format,
-                            par_format_name=par_format_name,
-        
-                            default_type=default_type,
-                            allowed_types=allowed_types,
-        
-                            check_value=check_value,
-                            allowed_values=allowed_values)
+                    parameter = x(**call_kwargs)
                     break
```

Comment:
I am a bit concerned what happens when there are multiple matches. In principle, it gets the most generic one, but maybe it's good to warn too.

Reply:
I think that we can just avoid multiple matches. Subclasses should have more specific uris than parents. 
What's exactly selected is related to https://github.com/oda-hub/dispatcher-app/issues/444

Reply:
we'll be dealing with these sort of challenges as the ontology improves, that's why I left todo's instead of invasive changes